src/components/Dropdown.tsx

```diff
@@ -9,9 +9,9 @@ import { getOption } from '../utils/getOption'
 
 type Props = {
   placeholder: string | null
-  options: DropdownOptionType[]
+  options: () => DropdownOptionType[]
```

Comment:
Should this use `Accessor`?

```suggestion
  options: Accessor<DropdownOptionType[]>
```